src/abstractdataframe/reshape.jl

```diff
@@ -355,12 +365,39 @@ julia> unstack(df, :variable, :value, fill=0)
 ─────┼──────────────────────
    1 │ 1           1      2
    2 │ 2           3      0
+
+julia> df = DataFrame(cols=["a", "a", "b"], values=[1, 2, 4])
+3×2 DataFrame
+ Row │ cols    values
+     │ String  Int64
+─────┼────────────────
+   1 │ a            1
+   2 │ a            2
+   3 │ b            4
+
+julia> unstack(df, :cols, :values, allowduplicates=identity)
+1×2 DataFrame
+ Row │ a       b
+     │ Array…  Array…
+─────┼────────────────
+   1 │ [1, 2]  [4]
+
+julia> unstack(df, :cols, :values, allowduplicates=sum)
+1×2 DataFrame
+ Row │ a      b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     3      4
 ```
-Note that there are some differences between the widened results above.
 """
 function unstack(df::AbstractDataFrame, rowkeys, colkey::ColumnIndex,
                  value::ColumnIndex; renamecols::Function=identity,
-                 allowmissing::Bool=false, allowduplicates::Bool=false, fill=missing)
+                 allowmissing::Bool=false,
+                 allowduplicates::Union{Bool, Function}=false, fill=missing)
+    if fill !== missing && allowduplicates isa Function
+        throw(ArgumentError("passing fill argument is not supported when " *
+                            "allowduplicates is a function"))
+    end
```

Comment:
So `missing` is used as a special sentinel value. I guess that's OK as we don't plan to allow passing `fill` when `allowduplicates isa Function`? I.e. the function can simply do `isempty(x) && return missing` for example, right?

Reply:
Right, I think that if user passes a `Function` it should get an empty vector and decide what to do with it. However, this is a case connected with https://github.com/JuliaData/DataFrames.jl/pull/2998#discussion_r796551634 comment and I will expand on the choice there.